### DISCUSSION TITLE
SvelteKit OAuth SSR

### DISCUSSION THREAD

**Discussion by svayp1337** [pocketbase/pocketbase#4771] [Q&A] [answered]
Can anyone please give me a simple solution for OAuth SSR on js with svelte?
All works fine with authWithPassword, but OAuth... works for me only with crutches in the code.

**Comment by orenaksakal** [comment, marked as answer]
not really SSR Oauth but all other parts of my app is SSR and this is what's been working for me nicely (including safari) 

OAuth with client side pocketbase client and set the auth cookie manually then refresh the page or redirect

```
on:click={async () => {
		loadingOAuth = true;

		let w = window.open();

		const authData = await pocketbaseBrowserClient.collection('users').authWithOAuth2({
			provider: 'google',
			urlCallback: (url) => {
				if (w) {
					w.location.href = url;
				}
			}
		});

		if (browser) {
			document.cookie = `pb_auth=${JSON.stringify({
				token: authData.token,
				model: authData.record
			})}; path=/`;
		}

		location.reload();
	}}
```

**Reply by svayp1337** [reply]
Looks very clean, thank you so much!
